File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.java
import java.util.ArrayList;

class Solution {
    boolean solution(String s) {
        boolean answer = true;
        ArrayList<Character> arr = new ArrayList<>();
        for (int i = 0; i < s.length(); i++) {
            if (s.charAt(i) == '(') {
                arr.add(s.charAt(i));
            } else {
                if (arr.size() == 0) {
                    answer = false;
                    break;
                } else {
                    arr.remove(arr.size() - 1);
                }
            }
        }
        if (arr.size() > 0) {
            answer = false;
        }
        System.out.println(answer);
        return answer;
    }
}

File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    arr = []
    answer = True
    for i in range(len(s)):
        if(s[i] == ')'):
            if(len(arr) == 0):
                answer = False
                break
            elif(arr.pop() == "("):
                continue
            else:
                print(arr.pop())
                answer = False
                break
        else:
            arr.append(s[i])
    if(len(arr) != 0 ):
        answer = False
    print(answer)
    return answer